modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/HashAggregateNode.java

```diff
@@ -77,11 +75,9 @@ public HashAggregateNode(
         this.accFactory = accFactory;
         this.rowFactory = rowFactory;
 
-        ImmutableBitSet.Builder b = ImmutableBitSet.builder();
+        assert grpSets.size() <= Byte.MAX_VALUE : "Too many grouping sets";
 
-        if (grpSets.size() > Byte.MAX_VALUE) {
-            throw new IgniteInternalException(TOO_MANY_GROUPING_EXPRESSIONS_ERR, "Too many groups");
-        }
+        ImmutableBitSet.Builder b = ImmutableBitSet.builder();
```

Comment:
Why did you changed an Exception to assertion?

Reply:
If we get here and an aggregate is invalid, we had an error at validation/planning stage.
If we want to keep an error here, we should also replace a lot of assertions in other ExecutionNodes with Exceptions as well.

Reply:
I tried to reproduce this case at the validation stage (to reject queries with more than 128 groups sets), but irregardless of how many grouping sets a query has, grpSets.size() is always 1.